Makefile

```diff
@@ -130,25 +183,29 @@ all: $(NAME)
 # Linking the NAME target
 $(NAME): $(LIBFT) $(OBJS)
 	@printf "\n$(YELLOW)$(BOLD)link binary$(RESET) [$(BLUE)miniRT$(RESET)]\n"
-	$(CC) $(LDFLAGS) $(OBJS) $(LDLIBS) -o $@
+	$(SILENT)$(CC) $(LDFLAGS) $(OBJS) $(LDLIBS) -o $@
 	@printf "\n$(YELLOW)$(BOLD)compilation successful$(RESET) [$(BLUE)miniRT$(RESET)]\n"
 	@printf "$(BOLD)$(GREEN)$(NAME) created!$(RESET)\n\n"
 
-# This target adds fsanitize leak checker to the flags. It needs to clean and recompile.
+# This target adds fsanitize leak checker to the flags.
 .PHONY: leak
 leak: CFLAGS += -fsanitize=leak
 leak: LDFLAGS += -fsanitize=leak
-leak: clean $(NAME)
-	@printf "Compiled with $(YELLOW)$(BOLD)fsanitize=leak$(RESET)\n\n"
+leak: $(NAME)
```

Comment:
No: We have different subdirectories for the differently compiled objects. E.g. if you compile "normal" objects (and depends) are in obj/default, if you compile for speed it gets put in obj/speed.
This increases the need for disk space (since we store both compiled versions of objects) but also makes compile times much faster if switching between targets

Reply:
I see, so we do not need to recompile when different flags are set